### ISSUE TITLE
impossible to use DerivedProperty with TimerNode

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/scenery-phet#426]
In Gas Properties, I'd like to keep track of time in seconds in the model, and display "scaled time" (1 s = 2.5 ps) in TimerNode.  I tried using a DerivedProperty to do this, and it worked nicely -- until I hit the reset button in TimerNode.   Unfortunately, TimerNode thinks that it knows what should be done when the reset buttons is pressed, and that is:

```js
      listener: function resetTimer() {
        runningProperty.set( false );
81      timeProperty.set( 0 );
      },
```

And when `timeProperty` is a DerivedProperty, line 81 fails. 

TimerNode would be much more flexible if it allowed the client to decide what should be done when the reset button is pressed.

**Comment by pixelzoom** [comment]
I went down the DynamicProperty path, and that required making both the DynamicProperty and the associated model Property `reentrant: true`.    I'm going to punt on this and change the model Property units to picoseconds, so that there's no difference between the model and view units.

So it's no longer needed in Gas Properties. But it sure seems like a general problem.  Back to @samreid to decide whether to address it, or wait until it comes up again.

**Comment by samreid** [comment]
> I'm going to punt on this and change the model Property units to picoseconds, so that there's no difference between the model and view units.

My recollection is that the developer team consensus is that models should typically be coded in the appropriate units for that model (no longer strictly MKS as was our prior convention), so this decision sounds appropriate.

**Comment by samreid** [comment]
> Back to @samreid to decide whether to address it, or wait until it comes up again.

Let's wait until it comes up again, unassigning until then.

**Comment by pixelzoom** [comment]
We had an extended discussion in https://github.com/phetsims/sun/issues/641 about the problems with the Property class hierarchy.  DerivedProperty extends Property, bu it's value is not directly settable.  An API may specify `@param {Property}`, but if it expects to be able to set that Property, then it cannot be a DerivedProperty.   

We decided to handle this by adding method `isSettable`, and assertion to check that `isSettable() === true` where a Property is expect to be settable.   I'm not sure what became of TimerNode in gas-properties (it no longer exists) so I'm not going to try to add such an assertion there. 

So nothing remains to be done here. Closing.
